src/App/Utilities/PasswordFormatter.cs

```diff
@@ -56,24 +54,39 @@ public static FormattedString FormatPassword(string password)
                 // If the char type changed, build a new span to append the text to.
                 if(charType != currentType)
                 {
-                    currentSpan = new Span();
-                    result.Spans.Add(currentSpan);
+                    // Close off previous span.
+                    if (currentType != CharType.None)
+                    {
+                        result += "</span>";
+                    }
+                    
                     currentType = charType;
 
                     // Switch the color if it is not a normal text. Otherwise leave the
                     // default value.
                     switch(currentType)
                     {
+                        // Apply color style to span.
+                        // First two digits of returned hex code contains the alpha,
+                        // which is not supported in HTML color,
+                        // so we need to cut those out.
                         case CharType.Number:
-                            currentSpan.TextColor = (Color)Application.Current.Resources["PasswordNumberColor"];
+                            result += $"<span style=\"color:#{((Color)Application.Current.Resources["PasswordNumberColor"]).ToHex().Substring(2)}\">";
                             break;
                         case CharType.Special:
-                            currentSpan.TextColor = (Color)Application.Current.Resources["PasswordSpecialColor"];
+                            result += $"<span style=\"color:#{((Color)Application.Current.Resources["PasswordSpecialColor"]).ToHex().Substring(2)}\">";
```

Comment:
Let's capture the hex codes as strings in variables outside of the for loop and then just use them here.